### ISSUE TITLE
Allow export/import of options

### ISSUE THREAD

**Issue by digital-carver** [Lej77/tab-unloader-for-tree-style-tab#37]
An export (and import) feature would be very useful to keep a backup of the settings, and to transfer them between different Firefox installations/profiles. Especially in the case of different profiles, even Firefox Sync is not an option since usually the point of different profiles is to keep them separate and not sync-ed up to the same Firefox account (in terms of tabs, other extensions, etc.). So for that use case, a json file or something that we can export when we change settings in one profile, and then import from another profile, would be the most flexible and useful feature to have.

**Comment by Lej77** [comment]
This is a good idea and I will see about implementing it.

**Comment by TFWol** [comment]
This really needs import/export.

There are at least 17 foldable headers and >110 checkboxes, not to mention the dropdowns and fields you can type in.

 It's _very_ tedious changing one option at a time.

**Comment by dsl101** [comment]
I'm trying to align browser behaviour across 4 machines, and it is still really tedious to check. Any chance this will get towards the top of a list?

Edit: As a workaround, this allowed me to move settings from one machine to another. Make sure you're running the same (latest) version of the add-on):

1. Navigate to `about:debugging#/runtime/this-firefox`
2. Find TST-Tab-Unloader and click Inspect
3. Open the console tab
4. Run `await browser.storage.local.get()`
5. Right-click on the result and click 'Copy Object'
6. Move to the target machine and repeat steps 1–3
7. Run `await browser.storage.local.set({<copied-object-here>})` making sure to paste the copied object inside the parentheses

Hope that's useful to someone :)

**Comment by Lej77** [comment]
Just released version `6.18` of this extension and it can now export and import the settings to and from a JSON file. It also added support for Firefox Sync though that code is a bit more complicated and untested so there might be bugs.

**Comment by dsl101** [comment]
I'm happy to test the sync; could you clarify exactly what all these options mean?

<img width="312" height="184" alt="Image" src="https://github.com/user-attachments/assets/c647460b-01a9-4aa5-b1ae-0a8c6d439060" />

Always and never is clear, but 'Follow global option' and that checkbox—are they directly connected? I.e. is the checkbox just a shortcut to either yes or no? Or is there some other 'global' option somewhere too?

**Comment by Lej77** [comment]
Yeah, so the idea is that you can enable sync if you want it (via the checkbox). Any new installation will default to the "Follow global option" setting which will sync if the checkbox is enabled. Note: the "Enable settings synchronization globally" checkbox is the only setting that will always be synced no matter your config.

If you want more control you can disable sync for a specific PC/browser by setting it to "Never sync". If you want new installations to not sync by default then just leave the checkbox disabled and go to those PCs where you want sync and set those to "Always sync".
